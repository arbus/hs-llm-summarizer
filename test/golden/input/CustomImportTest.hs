module CustomImportTest where

import PI.Db.User qualified as Db

-- | User data type that wraps around the database user
data User = User
  { userId   :: Int
  , userName :: String
  , userAge  :: Int
  }

-- | Convert database user to application user
fromDbUser :: Db.User -> User
fromDbUser dbUser = User
  { userId   = Db.id dbUser
  , userName = Db.name dbUser
  , userAge  = Db.age dbUser
  }

-- | Create a new user
createUser :: String -> Int -> User
createUser name age = User
  { userId   = 0  -- Will be assigned by database
  , userName = name
  , userAge  = age
  }