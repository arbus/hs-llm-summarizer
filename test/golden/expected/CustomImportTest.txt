-- Module: test/golden/input/CustomImportTest.hs

 | User data type that wraps around the database user

data User = User{userId :: Int, userName :: String, userAge :: Int}

 | Convert database user to application user

fromDbUser :: Db.User -> User



 | Create a new user

createUser :: String -> Int -> User

